src/librustc_mir/util/pretty.rs

```diff
@@ -441,18 +441,34 @@ fn write_scope_tree(
             continue;
         }
 
-        let indented_debug_info = format!(
-            "{0:1$}debug {2} => {3:?};",
-            INDENT, indent, var_debug_info.name, var_debug_info.place,
-        );
+        let mut indented_header = format!("{0:1$}debug {2}", INDENT, indent, var_debug_info.name);
+
+        match &var_debug_info.contents {
+            VarDebugInfoContents::Compact(place) => {
+                write!(indented_header, " => {:?};", place).unwrap();
+            }
+            VarDebugInfoContents::Composite { ty, fragments: _ } => {
+                write!(indented_header, ": {} {{", ty).unwrap();
```

Comment:
Should we include `fragments` here somehow (we do in `graphviz.rs` so it seems like we should have something here as well)?

Reply:
Look a bit further down ;).

(I even used the output from this in the PR description)